### PR TITLE
Action menu on list items

### LINKED ISSUE #622: Add ellipsis menu to list items in lime-elements

### PR DESCRIPTION
**Pull request by hannahu**
fix Lundalogik/crm-feature#989
### Browsers tested:

Windows:
- [ ] Chrome
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/list/list.tsx [hunk 1]
```diff
@@ -43,11 +47,18 @@ export class List {
     @Prop()
     public iconSize: IconSize = 'small';
 
+    /**
+     * Set to `true` if the list is open
+     */
+    @Prop()
+    public isOpen: boolean;
```
Comment (jgroth) [inline src/components/list/list.tsx]: Just adding a note here so I do not forget about it. We should investigate if we really need to add this, it's not very intuitive that a list has an `open` property. If we really need it, we might want to mark it as `internal` and add to the documentation that it's only used for the menu

File: src/components/menu/menu.tsx [hunk 2]
```diff
@@ -39,6 +30,12 @@ export class Menu {
     @Prop({ reflectToAttr: true })
     public disabled = false;
 
+    /**
+     * Decides if the menu should open right or left. Defaults to right.
+     */
+    @Prop({ reflectToAttr: true })
+    public openDirection: OpenDirection = 'right';
```
Comment (adrianschmidt) [inline src/components/menu/menu.tsx]: Defaulting to the right is a good choice I think, but it does kind of make this change breaking…

image

I'm fixing the above in Lundalogik/lime-webclient#1247

Comment (hannahu) [reply src/components/menu/menu.tsx]: There are more issues with the menu... opened an issue yesterday in the regular board:
https://github.com/Lundalogik/crm-feature/issues/1008

Comment (adrianschmidt) [reply src/components/menu/menu.tsx]: Ah, ok. Good that they've been caught 🙂

File: src/components/list/list-renderer.tsx [hunk 3]
```diff
@@ -70,7 +69,7 @@ export class ListRenderer {
         return (
             <ul
                 class={classNames}
-                aria-hidden={(!this.config.isOpen).toString()}
+                aria-hidden={(this.config.type === 'menu').toString()}
```
Comment (adrianschmidt) [inline src/components/list/list-renderer.tsx]: This doesn't seem quite right. Won't the menu list always be hidden from ARIA users?

File: src/components/menu/menu.tsx [hunk 4]
```diff
@@ -61,61 +58,39 @@ export class Menu {
...
-                {this.disabled ? <div class="menu-disabled" /> : null}
+                <limel-portal
+                    visible={this.open}
+                    containerId={this.portalId}
+                    openDirection={this.openDirection}
+                >
+                    <limel-menu-surface
+                        open={this.open}
+                        onDismiss={this.onClose}
+                    >
```
Comment (adrianschmidt) [inline src/components/menu/menu.tsx]: This version doesn't seem to be using a scrim, so how do we detect clicks outside the menu to make the menu close?

**Pull request merged**